File: android/src/main/java/com/evotor/egais/mapper/OrgInfo.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

private fun createOrgAddress(cursor: Cursor): WritableMap? {
    val country = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_COUNTRY"))
    val index = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_INDEX"))
    val regionCode = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_REGION_CODE"))
    val area = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_AREA"))
    val city = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_CITY"))
    val place = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_PLACE"))
    val street = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_STREET"))
    val house = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_HOUSE"))
    val building = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_BUILDING"))
    val liter = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_LITER"))
    val description = cursor.getString(cursor.getColumnIndexOrThrow("ADDRESS_DESCRIPTION"))
    if (country != null
            || index != null
            || regionCode != null
            || area != null
            || city != null
            || place != null
            || street != null
            || house != null
            || building != null
            || liter != null
            || description != null) {
        val result = Arguments.createMap()
        result.putString("country", country)
        result.putString("index", index)
        result.putString("regionCode", regionCode)
        result.putString("area", area)
        result.putString("city", city)
        result.putString("place", place)
        result.putString("street", street)
        result.putString("house", house)
        result.putString("building", building)
        result.putString("liter", liter)
        result.putString("description", description)
        return result
    }
    return null
}

internal fun createOrgInfo(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("type", cursor.getString(cursor.getColumnIndexOrThrow("TYPE")))
    result.putString("clientRegId", cursor.getString(cursor.getColumnIndexOrThrow("CLIENT_REG_ID")))
    result.putString("fullName", cursor.getString(cursor.getColumnIndexOrThrow("FULL_NAME_UPPER_CASE")))
    result.putString("shortName", cursor.getString(cursor.getColumnIndexOrThrow("SHORT_NAME_UPPER_CASE")))
    result.putString("inn", cursor.getString(cursor.getColumnIndexOrThrow("INN")))
    result.putString("kpp", cursor.getString(cursor.getColumnIndexOrThrow("KPP")))
    result.putString("unp", cursor.getString(cursor.getColumnIndexOrThrow("UNP")))
    result.putString("rnn", cursor.getString(cursor.getColumnIndexOrThrow("RNN")))
    result.putMap("address", createOrgAddress(cursor))
    result.putString("state", cursor.getString(cursor.getColumnIndexOrThrow("STATE")))
    result.putString("waybillVersion", cursor.getString(cursor.getColumnIndexOrThrow("VERSION_WB")))
    return result
}

File: model/dictionary/OrgInfo.js
/**
 * Классы и типы для работы с информацией об организациях.
 * @module module:orgInfo
 */
import {OrgType, WaybillTypeUsed} from "../types"

/**
 * @class module:orgInfo.OrgInfoAddress
 * @classdesc Фактический адрес произвольной формы.
 * @param {?string} country Страна
 * @param {?string} index Индекс
 * @param {?string} regionCode Код региона
 * @param {?string} area Область
 * @param {?string} city Город
 * @param {?string} place Район
 * @param {?string} street Улица
 * @param {?string} house Дом
 * @param {?string} building Строение
 * @param {?string} liter Литера
 * @param {?string} description Описание
 */
export class OrgInfoAddress {
    constructor(country: (string | null),
                index: (string | null),
                regionCode: (string | null),
                area: (string | null),
                city: (string | null),
                place: (string | null),
                street: (string | null),
                house: (string | null),
                building: (string | null),
                liter: (string | null),
                description: (string | null)) {
        this.index = index;
        this.regionCode = regionCode;
        this.area = area;
        this.city = city;
        this.place = place;
        this.street = street;
        this.house = house;
        this.building = building;
        this.liter = liter;
        this.description = description;
    }
}

/**
 * @class module:orgInfo.OrgInfo
 * @classdesc Информация об организации.
 * @param {?module:orgInfo#OrgType} type
 * @param {string} clientRegId Код организации
 * @param {string} fullName Наименование организации полное
 * @param {?string} shortName Наименование организации краткое
 * @param {?string} inn ИНН
 * @param {?string} kpp КПП
 * @param {?string} unp УНП для респ.Беларусь
 * @param {?string} rnn РНН(БИН/ИИН) для респ.Казахстан
 * @param {?module:orgInfo.OrgInfoAddress} address Фактический адрес
 * @param {?string} state Статус: действующая или удалена
 * @param {?module:orgInfo#WaybillTypeUsed} versionWB Версия схемы накладной принимаемая организацией
 */
export default class OrgInfo {
    constructor(type: (OrgType | null),
                clientRegId: string,
                fullName: string,
                shortName: (string | null),
                inn: (string | null),
                kpp: (string | null),
                unp: (string | null),
                rnn: (string | null),
                address: (OrgInfoAddress | null),
                state: (string | null),
                waybillVersion: (WaybillTypeUsed | null)) {
        this.type = type;
        this.clientRegId = clientRegId;
        this.fullName = fullName;
        this.shortName = shortName;
        this.inn = inn;
        this.kpp = kpp;
        this.unp = unp;
        this.rnn = rnn;
        this.address = address;
        this.state = state;
        this.waybillVersion = waybillVersion;
    }
}